### PR TITLE
Topic/content

### PR DESCRIPTION
**Pull request by GordanM**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+cmake_minimum_required(VERSION 3.2)
+
+# Project name
+project(template-library)
+
+# Project version
+set(PROJECT_MAJOR_VERSION 0)
+set(PROJECT_MINOR_VERSION 1)
+set(PROJECT_PATCH_LEVEL 0)
+set(PROJECT_VERSION ${PROJECT_MAJOR_VERSION}.${PROJECT_MINOR_VERSION}.${PROJECT_PATCH_LEVEL})
```
Comment (rpannek) [inline CMakeLists.txt]: Wouldn't it be nicer to set the version in the project() call like the documentation suggests? https://cmake.org/cmake/help/v3.0/command/project.html

File: include/templatepublicclass.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+class TemplatePublicClass
+{
+public:
+    TemplatePublicClass();
+    ~TemplatePublicClass();
+
+protected:
+
+private:
+    std::shared_ptr<TemplatePrivateClass> templatePrivateClass;
```
Comment (rpannek) [inline include/templatepublicclass.h]: Wouldn't it make more sense to call the variable `templatePrivateInstance` or `templatePrivateProperty`?

File: include/templatepublicclass.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#define TEMPLATEPUBLICCLASS_H_
+
+#include <memory>
+
+class TemplatePrivateClass;
+
+class TemplatePublicClass
+{
+public:
+    TemplatePublicClass();
```
Comment (rpannek) [inline include/templatepublicclass.h:22]: Perhaps you could add a `helloWorld()` method which would return a "hello world" string or something which we then could call from the service which uses this library?

**Review by rpannek — CHANGES_REQUESTED** [review]

**Comment by GordanM** [comment]
@rpannek thanks for the review. I've addressed your comments and a commit has been changed.

**Review by rpannek — APPROVED** [review]

**Comment by GordanM** [comment]
@breakreturn and @rpannek I have added the .pc.in file as suggested. Could you please check it?

**Review by rpannek — APPROVED** [review]
Looking good!

**Pull request merged**